### PR TITLE
Bug 1999092: Add and enable admin ack Upgradeable condition gate 

### PR DESCRIPTION
**Pull request by jottofar**
This PR backports "Bug 1978376: Add admin ack Upgradeable condition gate" and "pkg/cvo/upgradeable: Enable admin-ack logic". It also adds the 4.8 gate "ack-4.8-kube-122-api-removals-in-4.9" which blocks upgrades to 4.9 until an admin performs the required acknowledgement step.

The PR is composed of 3 commits the first 2 of which were created with cherry-pick:

$ git cherry-pick 519b466
$ git cherry-pick 5b5db7d

### CODE REVIEW

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml
Comment (wking) [inline install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:4]: nit: doesn't really matter, but with `4.8` and `4.9` both including an explicit `.`, it seems strange to use `122` for Kube instead of `1.22`.  Maybe change to `ack-4.8-kube-1.22-api-removals-in-4.9`?

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml
Comment (wking) [inline install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:6]: Hmm.  Do we have a UX for exposing these outside of the console and `oc` `Upgradeable` handling?  I'll check how the console handles linking, to see if we need a space between the URI and the trailing `.` or not...

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml
Comment (jottofar) [reply install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:4]: Agree. No reason that won't work and although that's what we specified in the enhancement that's shouldn't matter.

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml
Comment (bparees) [inline install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:6]: i hope we get to the point where we can point to better documentation for what an admin needs to go investigate/check before we ship this.

we should probably not merge this PR until we have that in place, since this gate will immediately start showing up in 4.8.z clusters.

**Comment by bparees** [comment]
/hold

we need the proper openshift documentation for what an admin is expected to do prior to ugprading to 4.9 (how to check api usage, how to ACK the gate, etc) before this merges, because once this merges anyone who upgrades to 4.8.z is going to start seeing this non-upgradeable condition.

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml
Comment (wking) [reply install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:6]: Turns out the lack of space between the URI and trailing `.` is fine; see the screenshot [here][1] (which is getting that condition linkified).

[1]: https://github.com/openshift/console/pull/9955#issue-984101602

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml
Comment (jottofar) [reply install/0000_00_cluster-version-operator_01_admingate_configmap.yaml]: changed

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml
Comment (jottofar) [inline install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:6]: Check the text pointing to the KCS.
/cc @sdodson
/cc @bparees 
/cc @wking

Comment (sdodson) [reply install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:6]: LGTM, @wking can you finish peer review and we'll unhold this after we confirm the publishing date with @bergerhoffer next week?

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml
Comment (bparees) [inline install/0000_00_cluster-version-operator_01_admingate_configmap.yaml:5]:
```suggestion
      Kubernetes 1.22 and therefore OpenShift 4.9 remove several APIs which require admin consideration. Please see
```

i dunno, "thus" stuck out to me, but maybe it's just me.

**Review by wking — APPROVED** [review]
With [cherry-pick-diff][1]:

```console
$ cherry-pick-diff origin/release-4.8..origin/pr/647 origin/release-4.9
fatal: bad revision 'origin/release-4.9^{/admin-gates. Add ack-4.8-kube-122-api-removals-in-4.9 gate}'
??? -> 9e21155d admin-gates: Add ack-4.8-kube-122-api-removals-in-4.9 gate
5b5db7db -> a5db8728 pkg/cvo/upgradeable: Enable admin-ack logic
519b4667 -> 61946347 Bug 1978376: Add admin ack Upgradeable condition gate
fatal: bad revision 'origin/release-4.9^{/admin-gates. Add ack-4.8-kube-122-api-removals-in-4.9 gate}'

5b5db7db -> a5db8728 pkg/cvo/upgradeable: Enable admin-ack logic
...unimportant context changes...
519b4667 -> 61946347 Bug 1978376: Add admin ack Upgradeable condition gate
...unimportant context changes, plus some *Operator vs. Operator changes because 4.8 doesn't have #603...
...and in 4.9, #637's etcd backup landed after #633's admin ack, while in 4.8 this admin ack is landing after #649's etcd backup...
```

The unique-to-4.8 9e21155d looks good to me.

/lgtm

[1]: https://github.com/openshift/cluster-version-operator/files/5882798/cherry-pick-diff.txt

File: install/0000_00_cluster-version-operator_01_admingate_configmap.yaml
Comment (jottofar) [reply install/0000_00_cluster-version-operator_01_admingate_configmap.yaml]: therefore, I changed it :)

Comment (bparees) [reply install/0000_00_cluster-version-operator_01_admingate_configmap.yaml]: thus, it looks good to me.

wait, what? :)

**Comment by sdodson** [comment]
/lgtm
String change only since last review, awaiting confirmation of publishing date.

**Comment by dhellmann** [comment]
[patch-manager] :hourglass: This pull request was not picked by the patch manager for the current z-stream window and have to wait for the next window.

*skipped for today*

* Score: *0.00*
* Reason: *skipping because "do-not-merge/hold" label found*

**NOTE**: This message was automatically generated, if you have questions please ask on #forum-release

**Comment by wking** [comment]
[`etcdHighNumberOfLeaderChanges`][1] is unrelated.

/override ci/prow/e2e-agnostic-upgrade

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/647/pull-ci-openshift-cluster-version-operator-release-4.8-e2e-agnostic-upgrade/1442842615597764608

**Pull request merged**
